Modules/ITS/src/ITSTrackTask.cxx

```diff
@@ -0,0 +1,222 @@
+#include <DataFormatsITS/TrackITS.h>
+#include <DataFormatsITSMFT/ROFRecord.h>
+
+using namespace o2::itsmft;
+using namespace o2::its;
+
+namespace o2::quality_control_modules::its
+{
+
+ITSTrackTask::ITSTrackTask() : TaskInterface()
+{
+
+  getEnableLayers();
+  createAllHistos();
+
+  o2::base::GeometryManager::loadGeometry();
+  mGeom = o2::its::GeometryTGeo::Instance();
+}
+
+ITSTrackTask::~ITSTrackTask()
+{
+
+  delete hNClusters;
+  delete hTrackEta;
+  delete hTrackPhi;
+  delete hOccupancyROF;
+  delete hClusterUsage;
+  delete hAngularDistribution;
+
+  delete mGeom;
+}
+
+void ITSTrackTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+
+  QcInfoLogger::GetInstance() << "initialize ITSTrackTask" << AliceO2::InfoLogger::InfoLogger::endm;
+  publishHistos();
+  std::string dictfile = o2::base::NameConf::getDictionaryFileName(o2::detectors::DetID::ITS, "", ".bin");
+  std::ifstream file(dictfile.c_str());
+
+  if (file.good()) {
+    LOG(INFO) << "Running with dictionary: " << dictfile.c_str();
+    mDict.readBinaryFile(dictfile);
+  } else {
+    LOG(INFO) << "Running without dictionary !";
+  }
+}
+
+void ITSTrackTask::startOfActivity(Activity& /*activity*/)
+{
+  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void ITSTrackTask::startOfCycle()
+{
+  QcInfoLogger::GetInstance() << "startOfCycle" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void ITSTrackTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+
+  QcInfoLogger::GetInstance() << "START DOING QC General" << AliceO2::InfoLogger::InfoLogger::endm;
+  auto trackArr = ctx.inputs().get<gsl::span<o2::its::TrackITS>>("tracks");
+  auto rofArr = ctx.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("rofs");
+  auto clusArr = ctx.inputs().get<gsl::span<o2::itsmft::CompClusterExt>>("compclus");
+
+  for (int iROF = 0; iROF < rofArr.size(); iROF++) {
+
+    auto& ROF = rofArr[iROF];
+    for (int itrack = ROF.getFirstEntry(); itrack < ROF.getFirstEntry() + ROF.getNEntries(); itrack++) {
+
+      auto& track = trackArr[itrack];
+
+      auto out = track.getParamOut();
+      Double_t Eta = -log(tan(out.getTheta() / 2.));
+      hTrackEta->Fill(Eta);
+      hTrackPhi->Fill(out.getPhi());
+      hAngularDistribution->Fill(Eta, out.getPhi());
+
+      hNClusters->Fill(track.getNumberOfClusters());
+      mNClustersInTracks += track.getNumberOfClusters();
+    }
+  }
+
+  mNTracks += trackArr.size();
+  mNClusters += clusArr.size();
+  mNRofs += rofArr.size();
+
+  if (mNRofs >= NROFOCCUPANCY) {
+    hOccupancyROF->SetBinContent(1, 1. * mNTracks / mNRofs);
+    mNTracks = 0;
+    mNRofs = 0;
+
+    hClusterUsage->SetBinContent(1, 1. * mNClustersInTracks / mNClusters);
+    mNClustersInTracks = 0;
+    mNClusters = 0;
+  }
+}
+
+void ITSTrackTask::endOfCycle()
+{
+
+  std::ifstream runNumberFile("/home/its/QC/workdir/infiles/RunNumber.dat"); //catching ITS run number in commissioning
```

Comment:
Why don't you do it at startOfActivity ?

Reply:
It's a workaround for our commissioning as the RunNumber file is created after the QC workflow already started. It would be completely different in production on how to get the ITS run number.

Reply:
ok